src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/multi-cluster/multi-cluster-list/multi-cluster-list.component.ts

```diff
@@ -162,15 +142,38 @@ export class MultiClusterListComponent extends ListWithDetails implements OnInit
         cellTemplate: this.durationTpl
       }
     ];
+    this.managedByConfig$ = this.settingsService.getValues('MANAGED_BY_CLUSTERS');
+  }
 
+  fetchData() {
     this.subs.add(
-      this.multiClusterService.subscribeClusterTokenStatus((resp: object) => {
-        this.clusterTokenStatus = resp;
-        this.checkClusterConnectionStatus();
-      })
-    );
+      zip(
+        this.multiClusterService.msData$.pipe(filter((value) => !!value)),
+        this.multiClusterService.tokenStatusSource$.pipe(filter((value) => !!value))
+      ).subscribe(
+        ([configResp, tokenStatusResp]: [any, any]) => {
+          this.clusterTokenStatus = tokenStatusResp;
 
-    this.managedByConfig$ = this.settingsService.getValues('MANAGED_BY_CLUSTERS');
+          if (configResp && configResp['config']) {
+            this.hubUrl = configResp['hub_url'];
+            this.currentUrl = configResp['current_url'];
+            const clusterDetailsArray = Object.values(configResp['config']).flat();
+            this.data = clusterDetailsArray;
+            this.checkClusterConnectionStatus();
+            this.data.forEach((cluster: any) => {
+              cluster['remainingTimeWithoutSeconds'] = 0;
+              if (cluster['ttl'] && cluster['ttl'] > 0) {
+                cluster['ttl'] = cluster['ttl'] * 1000;
+                cluster['remainingTimeWithoutSeconds'] = this.getRemainingTimeWithoutSeconds(
+                  cluster['ttl']
+                );
+                cluster['remainingDays'] = this.getRemainingDays(cluster['ttl']);
+              }
```

Comment:
lets break this into small readable functions which will help in improving readability of this code.

Thanks